demoapp/demoapp/models.py

```diff
@@ -0,0 +1,25 @@
+import hashlib
+
+from django.conf import settings
+from django.db import models
+from django.urls import reverse
+
+
+class SubscriberEmail(models.Model):
+    email = models.EmailField()
+
+    @property
+    def unsubscribe_token(self):
+        secret = settings.SECRET_KEY.encode()
+        unique = f"{self.id}{settings.SECRET_KEY}"
+        return hashlib.sha256(unique.encode()).hexdigest()[:32]
+
+    @property
+    def get_unsubscribe_url(self):
+        return reverse(
```

Comment:
This returns a relative URL which won't work IRL. I know this just the demo app but it's important that the whole thing makes sense.